### PR TITLE
Support collecting metrics of ZGC

### PR DESCRIPTION
**Pull request by xu1009**
### Add an agent plugin to support <zgc>
- [ ] Add a test case for the new plugin, refer to [the doc](https://github.com/apache/skywalking-java/blob/main/docs/en/setup/service-agent/java-agent/Plugin-test.md)
- [ ] Add a component id in [the component-libraries.yml](https://github.com/apache/skywalking/blob/master/oap-server/server-starter/src/main/resources/component-libraries.yml)
- [ ] Add a logo in [the UI repo](https://github.com/apache/skywalking-rocketbot-ui/tree/master/src/views/components/topology/assets)

- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<8411>.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking-java/blob/main/CHANGES.md).

### CODE REVIEW

File: apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/jvm/gc/CMSGCModule.java [hunk 1]
```diff
@@ -35,4 +35,9 @@ protected String getOldGCName() {
     protected String getNewGCName() {
         return "ParNew";
     }
+
+    @Override
+    protected String getNormalGcName() {
```
Comment (wu-sheng) [inline apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/jvm/gc/CMSGCModule.java:40]: Gc should be GC, it is a short term.

**Comment by xu1009** [comment]
wecom-temp-45c9bc374a0b653652ebb6999dbfd5df

File: apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/jvm/gc/CMSGCModule.java (hunk 1 above)
Comment (xu1009) [reply apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/jvm/gc/CMSGCModule.java]: ok

**Comment by xu1009** [comment]
ok

**Comment by xu1009** [comment]
it seems, the test not update submodule to compile and genrate code, so it fails

**Comment by wu-sheng** [comment]
Are you changing protocol back to old version by mistake?

**Comment by xu1009** [comment]
> Are you changing protocol back to old version by mistake?

no, i change GCPhrase to GCPhase follow the proto

**Comment by xu1009** [comment]
i get it

**Comment by wu-sheng** [comment]
Could you try this Java agent(with ZGC support) in ZGC JVM, but send data to OAP without ZGC support? I want to check whether the new added `normal` phase would work and what is the result?

**Comment by xu1009** [comment]
ok, i try it In the afternoon,  and feedback

**Comment by wu-sheng** [comment]
Thanks, you could add a breakpoint on JVM data deserialization, and check what value it will be in older proto data.

**Comment by xu1009** [comment]
wecom-temp-0757b9a96491765557364a18219cf844

wecom-temp-fe2d7e38e89136c139313c9a9585224d

image


the enum is unknown value when deserialization, the value is still 2

last the phase not record, and ui not display

File: CHANGES.md [hunk 2]
```diff
@@ -20,6 +20,7 @@ Release Notes.
 * Fix the bug that httpasyncclient-4.x-plugin  puts the dirty tracing context in the connection context
 * Compatible with the versions after dubbo-2.7.14
 * Follow protocol grammar fix `GCPhrase -> GCPhase`.
+* Support zgc gc time and count metric collect
```
Comment (wu-sheng) [inline CHANGES.md:23]:
```suggestion
* Support ZGC GC time and count metric collect. (Require 9.0.0 OAP)
```

I think it is better we add this. WDYT?

Comment (xu1009) [reply CHANGES.md:23]: ok

**Review by wu-sheng — APPROVED** [review]

File: apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/jvm/gc/GCProvider.java [hunk 3]
```diff
@@ -62,6 +62,9 @@ private GCMetricAccessor findByBeanName(String name) {
         } else if (name.equals("MarkSweepCompact")) {
             // Serial collector ( -XX:+UseSerialGC )
             return new SerialGCModule(beans);
+        } else if (name.indexOf("ZGC") > -1) {
+            // Serial collector ( -XX:+UseZGC )
```
Comment (kezhenxu94) [inline apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/jvm/gc/GCProvider.java:66]: @xu1009 can you open a PR to correct this comment?

```suggestion
            // Z garbage collector ( -XX:+UseZGC )
```

Comment (xu1009) [reply apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/jvm/gc/GCProvider.java:66]: ok

**Pull request merged**